File: solutions/2025-dec/31/markdown_italic_parser.js
function parseItalics(markdown) {
    let prev = null;
    const idx = [];
    for (let i = 0; i < markdown.length; i++) {
        if (prev === null && (markdown[i] == "*" || markdown[i] == "_")) {
            if (i + 1 < markdown.length && markdown[i + 1] != " ") prev = i;
        } else if (
            prev !== null &&
            (markdown[i] == "*" || markdown[i] == "_")
        ) {
            if (i - 1 >= 0 && markdown[i - 1] != " ") {
                idx.push(prev);
                idx.push(i);
                prev = null;
            }
        }
    }

    let ans = "";
    for (let i = 0; i < markdown.length; i++) {
        if (idx.includes(i)) {
            if (idx.indexOf(i) % 2 == 0) ans += "<i>";
            else ans += "</i>";
        } else {
            ans += markdown[i];
        }
    }
    return ans;
}

File: solutions/2025-dec/31/markdown_italic_parser.py
def parse_italics(markdown):
    prev = None
    idx = []
    for i in range(len(markdown)):
        if prev is None and markdown[i] in ("*", "_"):
            if i + 1 < len(markdown) and markdown[i + 1] != " ":
                prev = i
        elif prev is not None and markdown[i] in ("*", "_"):
            if i - 1 >= 0 and markdown[i - 1] != " ":
                idx.append(prev)
                idx.append(i)
                prev = None

    ans = ""
    for i in range(len(markdown)):
        if i in idx:
            if idx.index(i) % 2 == 0:
                ans += "<i>"
            else:
                ans += "</i>"
        else:
            ans += markdown[i]

    return ans


print(parse_italics("_ This is also not italic_"))
